modules/gmc-agent/src/main/java/org/gridgain/action/ActionDispatcher.java

```diff
@@ -45,6 +59,9 @@
      */
     public ActionDispatcher(GridKernalContext ctx) {
         this.ctx = ctx;
+
+        log = ctx.log(ActionDispatcher.class);
+        sesRegistry = SessionRegistry.getInstance(ctx);
```

Comment:
It is possible to implement SessionRegistry not as singleton?

Reply:
I think we can't, because we haven't the real DI container and can't inject any dependency in constructor :(